### PR TITLE
[feature fix] user/me route and Readme update

### PR DESCRIPTION
**Pull request by reinaH**
# Purpose: 
- https://github.com/CenterForOpenScience/osf.io/issues/3217 User detail information can be accessed by going to /users/{user_id}/. If a user is logged in, they can now access that information (and anything under) by going to users/me/. Additionally, if there are other places in the API that can use a user ID (for example, users/{user_id}/nodes), the same thing works there. 
- https://github.com/CenterForOpenScience/osf.io/issues/3053 API information was located in the API directory and is out of date. Moved that information into the main OSF Readme file and is now accurate

# Changes: 
- Removed API readme and edited main readme to contain correct information on how to run API.
- Made users/me route by editing code in get_user method in users/views.py: 

when user is logged in they can access user detail information by going to users/me/. Certain routes will and will not work (they have their respective unit tests): 

should work:
users/{user_id}
users/me
users/me/nodes
users/{user_id}/nodes

should not work:
users/{user_id}/me
users/{user_id}/nodes/me
nodes/me
nodes/{user_id}

# Side Effects:
None. 

# Notes: 
spoke to Brian G about including these 2 issues in a single pull request. Closes issues:
https://github.com/CenterForOpenScience/osf.io/issues/3217
https://github.com/CenterForOpenScience/osf.io/issues/3053

### CODE REVIEW

**Comment by brianjgeiger** [comment]
This PR still needs to follow our PR standards. https://cosdev.readthedocs.org/en/latest/process/pull_requests.html

**Comment by reinaH** [comment]
fixing it now.

**Comment by lyndsysimon** [comment]
This PR is an update of https://github.com/CenterForOpenScience/osf.io/pull/3240 - first pass was completed there. :point_up:

**Comment by brianjgeiger** [comment]
Below https://github.com/CenterForOpenScience/osf.io/pull/3246/files#diff-04c6e90faac2675aa89e2176d2eec7d8R140 put another bullet point just like it that says to `cp api/base/settings/local-dist.py api/base/settings/local.py`

File: api/README.md [hunk 1]
```diff
@@ -1,30 +0,0 @@
...
-# OSF API
-
-## Getting started
-
-From the root osf directory:
-
-```bash
-pip install -r requirements.txt
-# Required for browse-able API
-python manage.py collectstatic
```
Comment (abought) [inline api/README.md]: Is this still required for the web-browsable API tool? This requirement seems to have been dropped in the revised docs.

Comment (brianjgeiger) [inline api/README.md]: Yeaaah, I think it is.

File: tests/api_tests/users/test_views.py [hunk 2]
```diff
@@ -191,3 +191,145 @@ def test_get_projects_logged_in_as_different_user(self):
...
+        url = "/{}users/{}/nodes/me/".format(API_BASE, self.user_one._id)
+        res = self.app.get(url, auth=self.auth_one, expect_errors=True)
+        assert_equal(res.status_code, 404)
+
+    def test_get_404_path_users_user_id_nodes_me_user_not_logged_in(self):
+        url = "/{}users/{}/nodes/me/".format(API_BASE, self.user_one._id)
+        res = self.app.get(url, auth=self.auth_two, expect_errors=True)
+        assert_equal(res.status_code, 404)
+
+    def test_get_404_path_nodes_user_id_user_logged_in(self):
```
Comment (sloria) [inline tests/api_tests/users/test_views.py]: While these certainly don't hurt anything other than our test execution time, I'm not sure these tests are necessary (gasp). There are an infinite ways we can generate 404 responses--unless the test is providing useful verification of intended behavior, we can probably omit them.

That said, I think it's fine to leave these for now.

File: api/README.md (hunk 1 above)
Comment (sloria) [inline api/README.md]: @reinaH Why was this removed?

**Comment by sloria** [comment]
@reinaH There is a lot of noise in the commit history for this PR; a clean history makes future debugging easier. Can you please resend this PR with well-grouped commits and descriptive messages?

**Comment by reinaH** [comment]
@sloria brian g wanted me to delete the api readme and add the contents to the main readme

File: api/README.md (hunk 1 above)
Comment (reinaH) [inline api/README.md]: @sloria brian g wanted me to delete the api readme and add the contents to the main readme

Comment (brianjgeiger) [inline api/README.md]: That line should be kept. My fault; I forgot that it was a thing.

Comment (abought) [inline api/README.md]: Oh! I could have phrased my question better. I was referring specifically to the part about `python manage.py collectstatic`. Without it, the web-browsable API would look silly.

**Comment by sloria** [comment]
Closing this for now; feel free to reopen as a new PR

**Comment by brianjgeiger** [comment]
Did this get rebased after it was closed? It only has one commit attached.
